kernel/rros/poll.rs

```diff
@@ -0,0 +1,1201 @@
+use core::{
+    borrow::BorrowMut,
+    cell::{RefCell, RefMut},
+    mem::size_of,
+    ops::{Deref, DerefMut},
+    ptr::NonNull,
+    usize,
+};
+
+use crate::{
+    factory,
+    file::{
+        rros_get_file, rros_ignore_fd, rros_open_file, rros_put_file, rros_release_file,
+        rros_watch_fd, RrosFile,
+    },
+    flags::RrosFlag,
+    init_list_head,
+    list_add,
+    list_del,
+    list_empty,
+    list_entry,
+    list_entry_is_head,
+    list_first_entry,
+    list_for_each_entry,
+    list_for_each_entry_safe,
+    list_next_entry,
```

Comment:
why use list_head rather the list in the RFL?

Reply:
Thanks, I will try to change these `list_head` to `list` in the `RFL`.

Reply:
I have changed some of `list_head` to `List` in `RFL`. The rest of `list_head`s are difficult to change.